frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/backgroundables/BaseSlideInBackgroundable.java

```diff
@@ -86,11 +88,14 @@ public void run(ProgressIndicator indicator) {
       return;
     }
 
-    Try.run(() -> branchLayoutWriter.write(macheteFilePath, newBranchLayout, /* backupOldLayout */ true))
-        .onFailure(t -> VcsNotifier.getInstance(project).notifyError(/* displayId */ null,
-            /* title */ getString(
-                "action.GitMachete.BaseSlideInBelowAction.notification.title.branch-layout-write-fail"),
-            getMessageOrEmpty(t)));
+    ApplicationManager.getApplication()
+        .invokeLater(() -> WriteAction
+            .compute(() -> Try.run(() -> branchLayoutWriter.write(macheteFilePath, newBranchLayout, /* backupOldLayout */ true))
```

Comment:
and here - compute returns, run runs

Reply:
I'm not sure if we should just use plain `write` (which uses some plain `java.(n)io` under the hood)... this should probably happen via VFS. Same with reading the branch layout file. Lemme open an issue...

Reply:
This requirement to write files on UI thread is weird as hell, completely against how I'd intuitively see that 🤔

Reply:
https://github.com/VirtusLab/git-machete-intellij-plugin/issues/1227

In fact, while the write is realized by `java.nio.file.Files#write` (in `com.virtuslab.branchlayout.impl.readwrite.BranchLayoutFileWriter#write`), there's probably no need for `WriteAction` here; pls put a TODO for #1227 instead

Reply:
Huh, see my comments below; IMO #1303 needs to be done first before this PR is further progressed